### PR TITLE
fix: repair type error with BaseProvider::request, but actually

### PR DESCRIPTION
**Pull request by steveluscher**
Whoops! https://github.com/torusresearch/solana-embed/pull/23 was merged before I'd fully finished that PR. This is the rest of it!

### CODE REVIEW

**Comment by metallicalfa** [comment]
@steveluscher Thanks.

**Comment by metallicalfa** [comment]
@steveluscher Released solana-embed@0.1.0

File: src/embed.ts [hunk 1]
```diff
@@ -299,7 +299,7 @@ class Torus {
     if (!this.isInitialized) throw new Error("Torus is not initialized");
     const windowId = getWindowId();
     this.communicationProvider._handleWindow(windowId);
-    const topupResponse = await this.communicationProvider.request<boolean>({
+    const topupResponse = await this.communicationProvider.request<unknown, boolean>({
```
Comment (steveluscher) [inline src/embed.ts:302]: This was straight up laziness on my part. It should actually be the type of `{ provider, params, windowId }`.

Comment (metallicalfa) [reply src/embed.ts:302]: I'll make a PR for this one. Thanks

**Pull request merged**